transaction-status/src/parse_token/extension/confidential_transfer.rs

```diff
@@ -0,0 +1,399 @@
+                    ParseInstructionError::InstructionNotParsable(ParsableProgram::SplToken)
+                })?;
+            let amount: u64 = withdrawal_data.amount.into();
+            let proof_instruction_offset: i8 = withdrawal_data.proof_instruction_offset;
+            let mut value = json!({
+                "source": account_keys[account_indexes[0] as usize].to_string(),
+                "destination": account_keys[account_indexes[1] as usize].to_string(),
+                "mint": account_keys[account_indexes[2] as usize].to_string(),
+                "instructionsSysvar": account_keys[account_indexes[3] as usize].to_string(),
+                "amount": amount,
+                "decimals": withdrawal_data.decimals,
+                "newDecryptableAvailableBalance": format!("{}", withdrawal_data.new_decryptable_available_balance),
+                "proofInstructionOffset": proof_instruction_offset,
+
+            });
+            let map = value.as_object_mut().unwrap();
+            parse_signers(
+                map,
+                4,
+                account_keys,
+                account_indexes,
+                "owner",
+                "multisigOwner",
+            );
+            Ok(ParsedInstructionEnum {
+                instruction_type: "withdrawConfidentialTransfer".to_string(),
+                info: value,
+            })
+        }
+        ConfidentialTransferInstruction::Transfer => {
+            check_num_token_accounts(account_indexes, 5)?;
+            let transfer_data: TransferInstructionData = *decode_instruction_data(instruction_data)
+                .map_err(|_| {
+                    ParseInstructionError::InstructionNotParsable(ParsableProgram::SplToken)
+                })?;
+            let proof_instruction_offset: i8 = transfer_data.proof_instruction_offset;
+            let mut value = json!({
+                "source": account_keys[account_indexes[0] as usize].to_string(),
+                "destination": account_keys[account_indexes[1] as usize].to_string(),
+                "mint": account_keys[account_indexes[2] as usize].to_string(),
+                "instructionsSysvar": account_keys[account_indexes[3] as usize].to_string(),
+                "newSourceDecryptableAvailableBalance": format!("{}", transfer_data.new_source_decryptable_available_balance),
+                "proofInstructionOffset": proof_instruction_offset,
+
+            });
+            let map = value.as_object_mut().unwrap();
+            parse_signers(
+                map,
+                4,
+                account_keys,
+                account_indexes,
+                "owner",
+                "multisigOwner",
+            );
+            Ok(ParsedInstructionEnum {
+                instruction_type: "confidentialTransfer".to_string(),
+                info: value,
+            })
+        }
+        ConfidentialTransferInstruction::TransferWithFee => {
```

Comment:
This is separate from this PR, but now that I'm looking at this, what's the difference between `Transfer` and `TransferWithFee`? @samkim-crypto do we need more instruction data in `TransferWithFee` to convey the expected fee?

Reply:
Yeah, I wondered about those data structs being the same. I'll hold off on merging this until we hear from Sam.

Reply:
Right, the `Transfer` and `TransferWithFee` has the same data in the token program, but the corresponding `zk-token-proof` [instructions](https://github.com/solana-labs/solana/blob/master/zk-token-sdk/src/zk_token_proof_instruction.rs#L43-L62) are different. 

For `TransferWithFee`, if the expected fee is out in the clear, then the transfer amount can be deduced. So the expected fee is encrypted and then stuffed into the proof data since the proof program must verify the zkp to check that the expected fee was computed correctly.

Reply:
Do we need two different info structs, then? (`TransferInstructionData` and `TransferWithFeeInstructionData`)
ie. Is there a chance they might diverge in the future? Or can we DRY it now?

Reply:
Yes, good point. I think it makes sense to use a single struct here. Fee amount will always have to be in the proof instruction, so as long as the (non-confidential) `TransferChecked` and `TransferCheckedWithFee` remains to be just the fee amount, then the two confidential instructions data will remain the same as well.